crates/stark-felt/src/lib.rs

```diff
@@ -224,8 +303,136 @@ impl TryFrom<&Felt> for NonZeroFelt {
     }
 }
 
+impl FromPrimitive for Felt {
+    fn from_i64(value: i64) -> Option<Self> {
+        Self::from_i128(value as i128)
+    }
+
+    fn from_u64(value: u64) -> Option<Self> {
+        Self::from_u128(value as u128)
+    }
+
+    fn from_i128(value: i128) -> Option<Self> {
+        let mut res = Self(FieldElement::from(&UnsignedInteger::from(
+            value.unsigned_abs(),
+        )));
+        if value.is_negative() {
+            res = -res;
+        }
+        Some(res)
+    }
+
+    fn from_u128(value: u128) -> Option<Self> {
+        Some(Self(FieldElement::from(&UnsignedInteger::from(value))))
+    }
+}
+
+// TODO: we need to decide whether we want conversions to signed primitives
+// will support converting the high end of the field to negative.
+impl ToPrimitive for Felt {
+    fn to_u64(&self) -> Option<u64> {
+        self.to_u128().and_then(|x| u64::try_from(x).ok())
+    }
+
+    fn to_i64(&self) -> Option<i64> {
+        self.to_u128().and_then(|x| i64::try_from(x).ok())
+    }
+
+    fn to_u128(&self) -> Option<u128> {
+        match self.0.representative().limbs {
+            [0, 0, hi, lo] => Some((lo as u128) | ((hi as u128) << 64)),
+            _ => None,
+        }
+    }
+
+    fn to_i128(&self) -> Option<i128> {
+        self.to_u128().and_then(|x| i128::try_from(x).ok())
+    }
+}
+
+macro_rules! impl_from_u128 {
+    ($type:ty) => {
+        impl From<$type> for Felt {
+            fn from(value: $type) -> Self {
+                Self::from_u128(value as u128).expect("conversion from primitive is infallible")
```

Comment:
It's the simplest one. Is there any workflow where conversion is a bottleneck?

Reply:
I think it will be solved by implementing `From` first.

My idea was that maybe (I really have no idea), creating a Mont repr of a value lower than `u8::MAX` is cheaper than creating a Mont repr of a value greater than `u8::MAX`. And so on.

But it really depends on the internal of each team conversion to mont (which, again, I have no idea about)

Reply:
Right now I think lambdaworks doesn't differentiate for types. It may be a useful optimization to perform later, but it would require changes upstream first.

I already converted to use `From` and only implement `FromPrimitive` on top of it.